lib/api/core/storage/indexCache.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const
+  errorsManager = require('../../../config/error-codes/throw'),
+  Bluebird = require('bluebird');
+
+/**
+ * @deprecated
+ * Index/collection cache management
+ */
+class IndexCache {
+
+  constructor (kuzzle) {
+    this.kuzzle = kuzzle;
+
+    this.internalStorage = this.kuzzle.services.internalStorage;
+    this.publicStorage = this.kuzzle.services.publicStorage;
+
+    // RAM cache structure:
+    //   myIndex: {
+    //     scope: 'public',
+    //     collections: []
+    //   }
+    this.indexes = {};
+  }
+
+  /**
+   * Initializes the index cache with existing indexes and collections
+   *
+   * @returns {Promise}
+   */
+  async init () {
+    const promises = [];
+
+    promises.push(
+      this.internalStorage.listIndexes()
+        .then(indexes => this._addCollections(indexes, 'internal')));
+
+    promises.push(
+      this.publicStorage.listIndexes()
+        .then(indexes => this._addCollections(indexes, 'public')));
+
+    await Bluebird.all(promises);
+
+    await this._addCollectionsAliases();
+  }
+
+  /**
+   * Adds an index and/or a collection to the cache.
+   * An event is emitted for other cluster node synchronization if the cache
+   * has changed.
+   *
+   * @param {Object} arguments - index, collection, scope (public)
+   *
+   * @returns {boolean} True if the cache has changed
+   */
+  add ({ index, collection, scope='public' } = {}) {
+    let modified = false;
+
+    if (index && ! this.indexes[index]) {
+      this.indexes[index] = {
+        scope,
+        collections: []
+      };
+
+      modified = true;
+    }
+
+    if (collection && ! this.indexes[index].collections.includes(collection)) {
+      this.indexes[index].collections.push(collection);
+
+      modified = true;
+    }
+
+    if (modified) {
+      // @todo modify cluster for scope
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)